torch/distributed/_tensor/api.py

```diff
@@ -202,13 +213,9 @@ class DTensor(torch.Tensor):  # pyre-ignore[13]: pyre is bad at __new__
     def __new__(
         cls,
         local_tensor: torch.Tensor,
-        device_mesh: DeviceMesh,
-        placements: Tuple[Placement, ...],
+        spec: DTensorSpec,
```

Comment:
I think the main concern is if this `DTensor` constructor change can break some people's code (though they should not be calling it directly in the first place).

Quickly looking over the internal search (https://fburl.com/code/rqumbl0u), this seems like it should not be an issue.

Reply:
yeah I think I explicitly mentioned in the code comment that this is not a public API, so hopefully ppl did not use it ;) 
 https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/api.py#L216

Reply:
TorchTune is the unlucky one... The change did not make into pytorch 2.4, right? TorchTune FSDP2 assume pytorch 2.4. We might need to add `if-else` base on pytorch version in TorchTune (torchtune/utils/_distributed.py) https://github.com/pytorch/torchtune/pull/1087/files